anvil/sample.py

```diff
@@ -180,3 +182,30 @@ def tau_chain(_metropolis_hastings):
 
 def acceptance(_metropolis_hastings):
     return _metropolis_hastings[0][2]
+
+
+# TODO: figure out how to name each coupling block
+@torch.no_grad()
+def yield_configs_layerwise(loaded_model, base_dist, sample_size, layer_id):
+    v, _ = base_dist(sample_size)
+    if layer_id == 0:
+        return v
+
+    negative_mag = (v.sum(dim=1).sign() < 0).nonzero().squeeze()
+
+    i = 1
+    for block in loaded_model:
+        v, _ = block(v, 0, negative_mag)
+        # only want coupling layers
+        if len([tensor for tensor in block.state_dict().values()]) > 1:
+            if i == layer_id:
+                return v
+            else:
+                i += 1
```

Comment:
I will need to adjust this after #68 is merged, since the model is no longer necessarily a flat `Sequential` of coupling blocks.